### DISCUSSION TITLE
Adding Uno's Monaco Editor prevents app from running

### DISCUSSION THREAD

**Discussion by NotroDev** [unoplatform/uno#12292] [Q&A]
I'm creating an Uno Platform app by template with the same name.

I select "Default" with custom configuration: .NET 7.0 Platforms: macOS (catalyst), Windows, GTK, Linux Framebuffer (by the way, do I need the Linux Framebuffer for the Linux support..?) MVUX Unit Tests, UI Tests All extensions except Console, Serilog, Blank Fluent Theme

It runs fine when I debug on Windows. But, when I add https://github.com/unoplatform/uno.monaco-editor-uwp library to the Windows project, when Debugging there is this error:
```
Payload contains two or more files with the same target path "Uno.UI.Toolkit.dll". Source files:
C:\Users\Notro.nuget\packages\uno.winui\4.8.33\lib\net5.0-windows10.0.18362.0\Uno.UI.Toolkit.dll
C:\Users\Notro.nuget\packages\uno.ui\4.0.8\lib\netstandard2.0\Uno.UI.Toolkit.dll.
```
Okay, I know what it means. But I'm really confused by all of the UI "platforms" (i don't know how to name them xd) that Uno supports. Should I use other project template? Or what?

PS. Oops, I described adding Monaco to Windows project, but I should add it to base project, right? I tried it before, when I'm doing it I have compile errors. I'll provide more info tomorrow.

**Comment by jeromelaban** [comment]
Thanks for the report. The uno.monaco-editor-uwp package is not compatible with the windows project. You'll need to use the original package instead: https://github.com/hawkerm/monaco-editor-uwp.

**Reply by NotroDev** [reply]
> Thanks for the report. The uno.monaco-editor-uwp package is not compatible with the windows project. You'll need to use the original package instead: https://github.com/hawkerm/monaco-editor-uwp.

Okay, thanks!
Unfortunately, I can't even install the original package. I'm entering `Install-Package Monaco.Editor -Version 0.9.0-beta` command and it throws this error:
```
Install-Package: NU1202: Package Monaco.Editor 0.9.0-beta is not compatible with net6.0-windows10.0.19041 (.NETCoreApp, Version=v6.0). Package Monaco.Editor 0.9.0-beta supports: uap10.0 (UAP, Version=v10.0).
At line: 1 char: 1.
```
It's the same on Windows project, base project, and on .NET 6.0 and 7.0.

**Reply by jeromelaban** [reply]
I see! Indeed the Monaco package does not yet support WinUI and WebView 2. There's no workaround at this time that I know of.

**Reply by NotroDev** [reply]
> I see! Indeed the Monaco package does not yet support WinUI and WebView 2. There's no workaround at this time that I know of.

Oh, that's a pity... Thanks for help, tho!

**Reply by NotroDev** [reply]
> I see! Indeed the Monaco package does not yet support WinUI and WebView 2. There's no workaround at this time that I know of.

But wait, if uno.monaco-editor-uwp package doesn't support windows project, then what is it for?

**Reply by jeromelaban** [reply]
At this time, it's only for WebAssembly, using the UWP APIs. We'll modify it to support Wasm+WinUI APIs in the future, as we have Uno Playground updates coming.

**Reply by NotroDev** [reply]
> At this time, it's only for WebAssembly, using the UWP APIs. We'll modify it to support Wasm+WinUI APIs in the future, as we have Uno Playground updates coming.

Oh, okay, I understand. Thanks!
